src/cmd/flux-account-service.py

```diff
@@ -0,0 +1,423 @@
+
+            # handle a test-server.view_user request
+            handle.respond(msg, payload)
+        except sqlite3.OperationalError as db_err:
+            # an exception was raised in the view_user method
+            handle.respond_error(msg, db_err.errno, str(db_err))
+        except Exception as exc:
+            # fall through to a non-OSError exception
+            handle.respond_error(msg, 0, str(exc))
+
+    def add_user(self, handle, watcher, msg, arg):
+        try:
+            val = u.add_user(
+                self.conn,
+                msg.payload["username"],
+                msg.payload["bank"],
+                msg.payload["userid"],
+                msg.payload["shares"],
+                msg.payload["max_running_jobs"],
+                msg.payload["max_active_jobs"],
+                msg.payload["max_nodes"],
+                msg.payload["queues"],
+                msg.payload["projects"],
+            )
+
+            payload = {"add_user": val}
+
+            handle.respond(msg, payload)
+        except ValueError as val_err:
+            handle.respond_error(msg, val_err.errno, str(val_err))
+        except sqlite3.IntegrityError as integrity_err:
+            handle.respond_error(msg, integrity_err.errno, str(integrity_err))
+        except Exception as exc:
+            handle.respond_error(msg, 0, str(exc))
+
+    def delete_user(self, handle, watcher, msg, arg):
+        try:
+            val = u.delete_user(self.conn, msg.payload["username"], msg.payload["bank"])
+
+            payload = {"delete_user": val}
+
+            handle.respond(msg, payload)
+        except Exception as exc:
+            handle.respond_error(msg, 0, str(exc))
+
+    def edit_user(self, handle, watcher, msg, arg):
+        try:
+            val = u.edit_user(
+                self.conn,
+                msg.payload["username"],
+                msg.payload["bank"],
+                msg.payload["default_bank"],
+                msg.payload["shares"],
+                msg.payload["max_running_jobs"],
+                msg.payload["max_active_jobs"],
+                msg.payload["max_nodes"],
+                msg.payload["queues"],
+                msg.payload["projects"],
+                msg.payload["default_project"],
+            )
+
+            payload = {"edit_user": val}
+
+            handle.respond(msg, payload)
+        except ValueError as val_err:
+            handle.respond_error(msg, val_err.errno, str(val_err))
+        except Exception as exc:
+            handle.respond_error(msg, 0, str(exc))
+
+    def view_bank(self, handle, watcher, msg, arg):
+        try:
+            val = b.view_bank(
+                self.conn,
+                msg.payload["bank"],
+                msg.payload["users"],
+            )
+
+            payload = {"view_bank": val}
+
+            handle.respond(msg, payload)
+        except sqlite3.OperationalError as db_err:
+            handle.respond_error(msg, db_err.errno, str(db_err))
```

Comment:
`sqlite3.OperationalError` is a subclass of [sqlite3.Error](https://docs.python.org/3/library/sqlite3.html#sqlite3.Error), and does not appear to have an `errno` attribute.

I'd go through all of `flux-account-service.py` and check usage of exceptions to ensure attributes are correct. As noted above, also add some context to the error message, in case the string representation of the error doesn't indicate that it is a sqlite3 error.

Also, it might be good to consider how to test some of these error conditions, though I'm not sure how to do that.